pkg/virt-launcher/virtwrap/device/hostdevice/gpu/hostdev.go

```diff
@@ -79,9 +221,23 @@ func createHostDevicesMetadata(vmiGPUs []v1.GPU) []hostdevice.HostDeviceMetaData
 // The validation assumes that the assignment of a device to a specified GPU is correct,
 // therefore a simple quantity check is sufficient.
 func validateCreationOfAllDevices(gpus []v1.GPU, hostDevices []api.HostDevice) error {
-	if len(gpus) != len(hostDevices) {
+	gpusWithDP := []v1.GPU{}
+	gpusWithDRA := []v1.GPU{}
+	for _, gpu := range gpus {
+		if gpu.ClaimRequest != nil {
+			gpusWithDRA = append(gpusWithDRA, gpu)
+		} else {
+			gpusWithDP = append(gpusWithDP, gpu)
+		}
+	}
+
+	if len(gpusWithDRA) > 0 && len(gpusWithDP) > 0 {
+		return fmt.Errorf("invalid VMI spec, both DRA and device plugin GPUs are not supported at the same time")
+	}
+
+	if len(gpusWithDP) != len(hostDevices) {
 		return fmt.Errorf(
-			"the number of GPU/s do not match the number of devices:\nGPU: %v\nDevice: %v", gpus, hostDevices,
+			"the number of device plugin GPU/s do not match the number of devices:\nGPU: %v\nDevice: %v", gpusWithDP, hostDevices,
```

Comment:
WDYT about also validating DRA devices in the same way?
Also, maybe we can rename the function to emphasize it only validates DP devices?

Reply:
Added validation for DRA devices and changed the function